File: Factorial_Recursive.py


# prompts the user and inputs the number 
print("This program finds the factorial of a number.")
num1 = int(input("Which number would you like to input? "))

# creates a new variable to use in the loop
num2 = num1

# function takes in num2 and 
def fact(num2):
  if (num2 == 1):
    return 1
  return num2 * fact(num2 - 1)

num3 = fact(num2)

# prints the answer out  
print (f"The Factorial of", num1, "is", num3)

File: Factorial_Recursive.rb
# Program to find the factorial of a number recursively

# prompts the user and inputs the number 
print"This program finds the factorial of a number.  "
print "Which number would you like to input? "

num1 = gets.to_i


# function takes in num2 and returns a factorial
def fact(num1)
  num2 = num1
  if num2 < 0
    return "This is a negative number.\n  Please enter a positive number."
  if num2 <= 1
    1
  else 
    num2 * fact(num2 - 1)
  end
end
  


num3 = fact(num1)

# prints the answer out  
puts "The Factorial of" 
puts num1
puts "is"
puts num3

end
